cmd/containerboot/kube.go

```diff
@@ -19,22 +19,44 @@ import (
 	"tailscale.com/tailcfg"
 )
 
-// storeDeviceInfo writes deviceID into the "device_id" data field of the kube
-// secret secretName.
-func storeDeviceInfo(ctx context.Context, secretName string, deviceID tailcfg.StableNodeID, fqdn string, addresses []netip.Prefix) error {
-	// First check if the secret exists at all. Even if running on
-	// kubernetes, we do not necessarily store state in a k8s secret.
+// storeDeviceID writes deviceID to 'device_id' data field of the named
+// Kubernetes Secret.
+func storeDeviceID(ctx context.Context, secretName string, deviceID tailcfg.StableNodeID) error {
+	// First check if the Secret exists at all. Even if running on
+	// kubernetes, we don't necessarily store state in a Secret.
 	if _, err := kc.GetSecret(ctx, secretName); err != nil {
 		if s, ok := err.(*kube.Status); ok {
 			if s.Code >= 400 && s.Code <= 499 {
-				// Assume the secret doesn't exist, or we don't have
-				// permission to access it.
+				// Assume the Secret doesn't exist, or we don't
+				// have permissions to access it.
 				return nil
 			}
 		}
 		return err
 	}
```

Comment:
Maybe worth pulling this shared code out into a function like `func secretAccessible(ctx context.Context, secretName string) (bool, error)`?

Reply:
Thinking about it, looks like this check is now redundant. We check that we have the right permissions to get/patch/create Secret [here](https://github.com/tailscale/tailscale/blob/v1.69.0-pre/cmd/containerboot/main.go#L192). I am going to remove this bit altogether.

Reply:
Nice! 🎉

Reply:
(did some additional testing to ensure that in cases, where the Secret is not precreated (and is expected to be created by kube state store used by tailscaled), this does not race with Secret creation)